src/thread/libcap.rs

```diff
@@ -0,0 +1,189 @@
+        /// `CAP_SYS_RESOURCE`
+        const SYS_RESOURCE = 1 << linux_raw_sys::general::CAP_SYS_RESOURCE;
+        /// `CAP_SYS_TIME`
+        const SYS_TIME = 1 << linux_raw_sys::general::CAP_SYS_TIME;
+        /// `CAP_SYS_TTY_CONFIG`
+        const SYS_TTY_CONFIG = 1 << linux_raw_sys::general::CAP_SYS_TTY_CONFIG;
+        /// `CAP_MKNOD`
+        const MKNOD = 1 << linux_raw_sys::general::CAP_MKNOD;
+        /// `CAP_LEASE`
+        const LEASE = 1 << linux_raw_sys::general::CAP_LEASE;
+        /// `CAP_AUDIT_WRITE`
+        const AUDIT_WRITE = 1 << linux_raw_sys::general::CAP_AUDIT_WRITE;
+        /// `CAP_AUDIT_CONTROL`
+        const AUDIT_CONTROL = 1 << linux_raw_sys::general::CAP_AUDIT_CONTROL;
+        /// `CAP_SETFCAP`
+        const SETFCAP = 1 << linux_raw_sys::general::CAP_SETFCAP;
+        /// `CAP_MAC_OVERRIDE`
+        const MAC_OVERRIDE = 1 << linux_raw_sys::general::CAP_MAC_OVERRIDE;
+        /// `CAP_MAC_ADMIN`
+        const MAC_ADMIN = 1 << linux_raw_sys::general::CAP_MAC_ADMIN;
+        /// `CAP_SYSLOG`
+        const SYSLOG = 1 << linux_raw_sys::general::CAP_SYSLOG;
+        /// `CAP_WAKE_ALARM`
+        const WAKE_ALARM = 1 << linux_raw_sys::general::CAP_WAKE_ALARM;
+        /// `CAP_BLOCK_SUSPEND`
+        const BLOCK_SUSPEND = 1 << linux_raw_sys::general::CAP_BLOCK_SUSPEND;
+        /// `CAP_AUDIT_READ`
+        const AUDIT_READ = 1 << linux_raw_sys::general::CAP_AUDIT_READ;
+        /// `CAP_PERFMON`
+        const PERFMON = 1 << linux_raw_sys::general::CAP_PERFMON;
+        /// `CAP_BPF`
+        const BPF = 1 << linux_raw_sys::general::CAP_BPF;
+        /// `CAP_CHECKPOINT_RESTORE`
+        const CHECKPOINT_RESTORE = 1 << linux_raw_sys::general::CAP_CHECKPOINT_RESTORE;
+    }
+}
+
+/// `capget(_LINUX_CAPABILITY_VERSION_3, pid)`
+///
+/// # References
+///  - [Linux]
+///
+/// [Linux]: https://man7.org/linux/man-pages/man2/capget.2.html
+#[inline]
+pub fn get_capabilities(pid: Option<Pid>) -> io::Result<CapabilitySets> {
+    capget(pid)
+}
+
+/// `capset(_LINUX_CAPABILITY_VERSION_3, pid, effective, permitted, inheritable)`
+///
+/// # References
+///  - [Linux]
+///
+/// [Linux]: https://man7.org/linux/man-pages/man2/capget.2.html
+#[inline]
+pub fn set_capabilities(pid: Option<Pid>, sets: CapabilitySets) -> io::Result<()> {
+    capset(pid, sets)
+}
+
+#[inline]
+#[allow(unsafe_code)]
+fn capget(pid: Option<Pid>) -> io::Result<CapabilitySets> {
+    let mut data = MaybeUninit::<[linux_raw_sys::general::__user_cap_data_struct; 2]>::uninit();
+
+    let data = {
+        let mut header = linux_raw_sys::general::__user_cap_header_struct {
+            version: linux_raw_sys::general::_LINUX_CAPABILITY_VERSION_3,
+            pid: Pid::as_raw(pid) as backend::c::c_int,
+        };
+
+        backend::thread::syscalls::capget(
+            &mut header as *mut linux_raw_sys::general::__user_cap_header_struct,
+            &mut data as *mut _ as *mut linux_raw_sys::general::__user_cap_data_struct,
```

Comment:
Instead of doing a pointer cast, can this use [`as_mut_ptr()`](https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.as_mut_ptr)?